config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/ServiceEntityBuilder.java

```diff
@@ -45,21 +46,22 @@
 
     public List<Entity> build(Bundle bundle, BundleType bundleType, Document document) {
         if (bundle instanceof AnnotatedBundle) {
+            AnnotatedBundle annotatedBundle = ((AnnotatedBundle) bundle);
             Map<String, Service> serviceMap = Optional.ofNullable(bundle.getServices()).orElse(Collections.emptyMap());
-            return buildEntities(serviceMap, ((AnnotatedBundle)bundle).getFullBundle(), bundleType, document);
+            return buildEntities(serviceMap, annotatedBundle, annotatedBundle.getFullBundle(), bundleType, document);
         } else {
-            return buildEntities(bundle.getServices(), bundle, bundleType, document);
+            return buildEntities(bundle.getServices(), null, bundle, bundleType, document);
         }
```

Comment:
We don't need RawBundle here as we have already found the entities going in the Annotated bundle. So this method can be simplified to:
`return buildEntities(bundle.getServices(), bundle, bundleType, document);`
In the `buildEntities()` you can check using `bundle instanceof AnnotatedBundle`

Reply:
done